public/pages/ShrinkIndex/container/ShrinkIndex/ShrinkIndex.tsx

```diff
@@ -0,0 +1,648 @@
+        this.context.notifications.toasts.addDanger(result?.error || errorMessage);
+      }
+    } catch (err) {
+      this.context.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem updating index settings."));
+    }
+  };
+
+  getAlias = async (aliasName: string) => {
+    try {
+      const { commonService } = this.props;
+      return await commonService.apiCaller<{ alias: string }[]>({
+        endpoint: "cat.aliases",
+        method: "GET",
+        data: {
+          format: "json",
+          name: `${aliasName || ""}*`,
+          expand_wildcards: "open",
+        },
+      });
+    } catch (err) {
+      this.context.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem getting aliases."));
+    }
+  };
+
+  openIndex = async (index: string) => {
+    try {
+      const { commonService } = this.props;
+      const result = await commonService.apiCaller({
+        endpoint: "indices.open",
+        data: {
+          index: index,
+        },
+      });
+      if (result && result.ok) {
+        this.context.notifications.toasts.addSuccess(`[${index}] has been set to Open.`);
+      } else {
+        this.context.notifications.toasts.addDanger(result.error);
+      }
+    } catch (err) {
+      this.context.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem opening index."));
+    }
+  };
+
+  isSourceIndexReady = async () => {
+    const { sourceIndex } = this.state;
+    const indexSettings = await this.getIndexSettings(sourceIndex.index, true);
+    this.setState({
+      sourceIndexSettings: indexSettings,
+    });
+  };
+
+  onUpdateIndexSettings = async (indexName: string, settings: {}) => {
+    await this.setIndexSettings(indexName, settings);
+
+    // refresh index settings
+    const indexSettings = await this.getIndexSettings(indexName, true);
+    this.setState({
+      sourceIndexSettings: indexSettings,
+    });
+  };
+
+  onOpenIndex = async () => {
+    const { sourceIndex } = this.state;
+    await this.openIndex(sourceIndex.index);
+
+    // refresh status
+    await this.getIndex(sourceIndex.index);
+  };
+
+  render() {
+    const { sourceIndex } = this.state;
+    if (!sourceIndex.index) {
+      return (
+        <EuiFlexGroup justifyContent="center" alignItems="center" style={{ marginTop: "100px" }}>
+          <EuiLoadingSpinner size="xl" />
+        </EuiFlexGroup>
+      );
+    }
+    const { requestPayload, sourceIndexSettings } = this.state;
+    const sourceIndexNotReadyToShrinkReasons: React.ReactChild[] = [];
+    let disableShrinkButton = false;
+    const blockNameList = ["targetIndex"];
+
+    if (sourceIndex.health === "red") {
+      disableShrinkButton = true;
+      sourceIndexNotReadyToShrinkReasons.push(
+        <>
+          <EuiCallOut
+            title="The source index's health status is Red, please check its status before shrinking."
+            color="danger"
+            iconType="alert"
+          ></EuiCallOut>
```

Comment:
how about this message ```The index is currently in the Red health state and may be running operations in the background. Wait until the index becomes Green to continue shrinking.```

Reply:
But there are many reasons why the index is Red, like a node was left from the cluster and has not been back, or one primary shard's segment files are corrupted, so running operations maybe not a common reason, I think. Thanks for your advice, I'will think over the wording.